pylib/zeus/zeus.py

```diff
@@ -30,15 +30,20 @@ def find_zeus_executable(current_path):
   :return: Path to the executable or None if not found.
   """
 
+  wanted_steps = (None, None)
   while True:
+    wanted_steps = (wanted_steps[1], os.path.basename(os.path.normpath(current_path)))
     # Check all files in the current directory
-    for file in os.listdir(current_path):
-      # Check if the file starts with 'zeus_' and has no extension
-      if file.startswith('zeus_') and '.' not in file:
-        file_path = os.path.join(current_path, file)
-        # Check if the file is executable
-        if os.access(file_path, os.X_OK):
-          return file_path
+    list_dir = frozenset(os.listdir(current_path))
+    for wanted_step in wanted_steps:
+        if wanted_step is None:
+            continue
+        file_name = 'zeus_' + wanted_step
```

Comment:
Doing this means we need to update the comment above. We're no longer looking for "zeus_" - we're specifically looking for either "zeus_<name of current folder>" or "zeus_<name of folder we just descended down from>" (which again - I don't understand why the child folder we came from matters?)

Reply:
it still checks files in current directory though?